From decf8d9a5f91912768d52cd075b6a25617a90d1c Mon Sep 17 00:00:00 2001
From: Jan Kratochvil <jan.kratochvil@redhat.com>
Date: Thu, 2 Jul 2015 22:39:57 +0200
Subject: [PATCH] Fix GCC false warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

At least on
	gcc-4.4.7-11.el6.i686
	./configure --enable-64-bit-bfd --enable-targets=all
GDB does not build due to:
	cc1: warnings being treated as errors
	s390-linux-tdep.c: In function ‘s390_handle_arg’:
	s390-linux-tdep.c:2575: error: ‘val’ may be used uninitialized in this function

gdb/ChangeLog
2015-07-02  Jan Kratochvil  <jan.kratochvil@redhat.com>

	Fix GCC false warning.
	* s390-linux-tdep.c (s390_handle_arg): Initialize VAL.
---
 gdb/ChangeLog         | 5 +++++
 gdb/s390-linux-tdep.c | 3 ++-
 2 files changed, 7 insertions(+), 1 deletion(-)

### a/gdb/ChangeLog
### b/gdb/ChangeLog
## -1,3 +1,8 @@
+2015-07-02  Jan Kratochvil  <jan.kratochvil@redhat.com>
+
+	Fix GCC false warning.
+	* s390-linux-tdep.c (s390_handle_arg): Initialize VAL.
+
 2015-07-02  Yao Qi  <yao.qi@linaro.org>
 
 	* aarch64-linux-nat.c (aarch64_linux_insert_hw_breakpoint): Fix
--- a/gdb/s390-linux-tdep.c
+++ b/gdb/s390-linux-tdep.c
@@ -2572,7 +2572,8 @@ s390_handle_arg (struct s390_arg_state *as, struct value *arg,
     }
   else if (s390_function_arg_integer (type) && length <= word_size)
     {
-      ULONGEST val;
+      /* Initialize it just to avoid a GCC false warning.  */
+      ULONGEST val = 0;
 
       if (write_mode)
 	{
-- 
2.1.0

